### PR TITLE
Use to_flyte_literal_type() when comparing schema columns

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
A user internally wanted to subclass one of our primitive types, and then created a separate type engine, and used said engine to resolve types.  However, since the Datetime primitive type was subclassed, the comparison failed.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
This relegates the comparison to a common denominator, namely the Flyte literal type object.

## Tracking Issue
Internal issue only

## Follow-up issue
NA

### CODE REVIEW

File: flytekit/common/types/impl/schema.py [hunk 1]
```diff
@@ -947,7 +947,7 @@ def cast_to(self, other_type):
                         additional_msg="Cannot cast because a required column '{}' was not found.".format(k),
                         received_value=self
                     )
-                if v != self.type.sdk_columns[k]:
```
Comment (kumare3) [inline flytekit/common/types/impl/schema.py:950]: should you first not check that isinstance(v, SdkType)?

Comment (wild-endeavor) [reply flytekit/common/types/impl/schema.py:950]: sure can you re-approve?

File: tests/flytekit/unit/common_tests/types/test_schema.py [hunk 2]
```diff
@@ -52,3 +52,20 @@ def test_typed_schema():
         assert len(b.type.columns) == len(_ALL_COLUMN_TYPES)
         assert list(b.type.sdk_columns.items()) == _ALL_COLUMN_TYPES
         assert b.remote_location.startswith(t.name)
+
+
+# Ensures that subclassing types works inside a schema.
+def test_casting():
+    class MyDateTime(primitives.Datetime):
+        ...
```
Comment (matthewphsmith) [inline tests/flytekit/unit/common_tests/types/test_schema.py:60]: this works? `...` is an allowed syntax?

Comment (wild-endeavor) [reply tests/flytekit/unit/common_tests/types/test_schema.py:60]: after i disabled travis's 2.7 tests, yup.
